examples/chef/common/chef-fan-control-manager.cpp

```diff
@@ -0,0 +1,176 @@
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/clusters/fan-control-server/fan-control-server.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/error-mapping.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::FanControl;
+using namespace chip::app::Clusters::FanControl::Attributes;
+using Protocols::InteractionModel::Status;
+
+namespace {
+class ChefFanControlManager : public AttributeAccessInterface, public Delegate
+{
+public:
+    ChefFanControlManager(EndpointId aEndpointId) :
+        AttributeAccessInterface(Optional<EndpointId>(aEndpointId), FanControl::Id), Delegate(aEndpointId)
+    {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    Status HandleStep(StepDirectionEnum aDirection, bool aWrap, bool aLowestOff) override;
+
+private:
+    CHIP_ERROR ReadPercentCurrent(AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadSpeedCurrent(AttributeValueEncoder & aEncoder);
+};
+
+static ChefFanControlManager * mFanControlManager = nullptr;
+
+CHIP_ERROR ChefFanControlManager::ReadPercentCurrent(AttributeValueEncoder & aEncoder)
+{
+    // Return PercentSetting attribute value for now
+    DataModel::Nullable<Percent> percentSetting;
+    EmberAfStatus status = PercentSetting::Get(mEndpoint, percentSetting);
+
+    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == status, CHIP_ERROR_READ_FAILED);
+
+    return aEncoder.Encode(percentSetting.ValueOr(0));
+}
+
+CHIP_ERROR ChefFanControlManager::ReadSpeedCurrent(AttributeValueEncoder & aEncoder)
+{
+    // Return SpeedCurrent attribute value for now
+    DataModel::Nullable<uint8_t> speedSetting;
+    EmberAfStatus status = SpeedSetting::Get(mEndpoint, speedSetting);
+
+    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == status, CHIP_ERROR_READ_FAILED);
+
+    return aEncoder.Encode(speedSetting.ValueOr(0));
+}
+
+Status ChefFanControlManager::HandleStep(StepDirectionEnum aDirection, bool aWrap, bool aLowestOff)
+{
+    ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleStep aDirection %d, aWrap %d, aLowestOff %d",
+                    to_underlying(aDirection), aWrap, aLowestOff);
+
+    VerifyOrReturnError(aDirection != StepDirectionEnum::kUnknownEnumValue, Status::InvalidCommand);
+
+    EmberAfStatus status;
+
+    uint8_t speedMax;
+    status = SpeedMax::Get(mEndpoint, &speedMax);
+    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == status, Status::InvalidCommand);
+
+    DataModel::Nullable<uint8_t> speedSetting;
+    status = SpeedSetting::Get(mEndpoint, speedSetting);
+    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == status, Status::InvalidCommand);
+
+    uint8_t newSpeedSetting = speedSetting.ValueOr(0);
+    const uint8_t kLowest   = aLowestOff ? 0 : 1;
```

Comment:
probably kLowestSpeed as a name to know 'lowest what'

Reply:
OK. Changed to `kLowestSpeed`